### PR TITLE
Fixes #20294,#20259 - taxonomy via fact fixed, scripts

### PR DESCRIPTION
**Pull request by lzap**
I have an improved script to do ad-hoc fake hosts discovery. It will become part of the main git repository.

Old scripts can be removed, they are not used anymore (this was from first generation discovery).

### CODE REVIEW

**Comment by lzap** [comment]
You will find this useful @rahulbajaj0509 :-) Please review.

**Comment by rabajaj0509** [comment]
@lzap , the script was indeed informative :)

Concerned with one thing here:
The fake discovered host is created in the organization that is alphabetically first among all the organizations created and does not get created in the organization i want, is this the correct functionality ?

**Comment by lzap** [comment]
Yeah, I can perhaps add custom org/loc option, lemme do that. Unfortunately the way it works is super-clunky, read this: https://theforeman.org/plugins/foreman_discovery/9.1/index.html#3.1.3OrganizationsandLocations

**Comment by lzap** [comment]
Try now! New options org/loc are present. Oh when I was testing this it was not working, reported a bug and fixed it here:

http://projects.theforeman.org/issues/20294

**Comment by rabajaj0509** [comment]
@lzap Thanks 👍  looks good to me !

**Comment by lzap** [comment]
Adding a test, @dLobatog this would be also in 9.1.2 if possible.

File: test/unit/host_discovered_test.rb [hunk 1]
```diff
@@ -42,6 +42,18 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert_equal subnet, host.primary_interface.subnet
   end
 
+  test "should setup subnet with org and loc set via settings" do
```
Comment (dLobatog) [inline test/unit/host_discovered_test.rb]: Please rename this one, two tests have the same name so only one runs I think

**Review by dLobatog — CHANGES_REQUESTED** [review]

File: test/unit/host_discovered_test.rb (hunk 1 above)
Comment (lzap) [reply test/unit/host_discovered_test.rb]: Done

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @lzap !

**Pull request merged**